File: Python/Array/E-169. Majority Element.py
class Solution:
    @classmethod
    def majorityElement(self, nums) -> int:
        dict = {}
        threshold = len(nums) / 2
        for num in nums:
            count = (dict.get(num) or 0) + 1
            if count > threshold:
                return num
            dict[num] = count


test_data = [
    [3, 2, 3],
    [2, 2, 1, 1, 1, 2, 2],
]
for data in test_data:
    print(Solution.majorityElement(data))

File: Array/E-169. Majority Element/E-169. Majority Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  const obj = {};
  const threshold = nums.length / 2;
  for (const num of nums) {
    const count = (obj[num] || 0) + 1;
    if (count > threshold) return num;
    obj[num] = count;
  }
};

const testData = [
  [3, 2, 3],
  [2, 2, 1, 1, 1, 2, 2],
];

for (const data of testData) {
  console.log(majorityElement(data));
}
